src/ua_securechannel.h

```diff
@@ -96,6 +96,9 @@ struct UA_SecureChannel {
     const UA_SecurityPolicy *securityPolicy;
     void *channelContext; /* For interaction with the security policy */
     UA_Connection *connection;
+    UA_CertificateVerification *certificateVerification;
```

Comment:
The asymmetric header is forwarded to the client/server in the new configure callback.
The verification could be done inside the configure callback.
(Simpler structures are better structures...)

Reply:
I'd rather have a pointer to the certVer in the securechannel than duplicate code to verify the certificate in client and server.
The certificate verification step is an essential part of the securechannel imho and should be located there.

Choosing the correct security policy on the other hand is server specific behavior, and located in the configure call.
On the client side we need to do next to nothing.
Currently only the flag that the handshake is complete is set.